frappe/utils/global_search.py

```diff
@@ -373,11 +373,57 @@ def sync_global_search():
 	:param flags:
 	:return:
 	"""
-	while frappe.cache.llen("global_search_queue") > 0:
-		# rpop to follow FIFO
-		# Last one should override all previous contents of same document
-		value = json.loads(frappe.cache.rpop("global_search_queue").decode("utf-8"))
-		sync_value(value)
+	from itertools import islice
+
+	def get_search_queue_item_generator():
+		while frappe.cache.llen("global_search_queue") > 0:
+			yield frappe.cache.rpop("global_search_queue")
+
+	item_generator = get_search_queue_item_generator()
+	while search_items := tuple(islice(item_generator, 10_000)):
+		values = get_deduped_search_item_values(search_items)
+		sync_values(values)
+
+
+def get_deduped_search_item_values(items):
```

Comment:
```suggestion
def _get_deduped_search_item_values(items):
```

If things are in `utils/*` prefer prefixing internal functions with `_`, so they can be removed/modified without worrying about side effects.